### ISSUE TITLE
Teku fails if deposit snapshots cannot be loaded

### ISSUE THREAD

**Issue by yorickdowne** [Consensys-Incorporated/teku#7827]
### Description

If Teku 23.12.0 is unable to get the deposit snapshots from the checkpoint sync provider, e.g. because that provider is unreachable, it fails to start.

Logs
```
eth-docker-devel-consensus-1  | 2023-12-13 22:35:38.055 INFO  - Listening on http://0.0.0.0:5052/
eth-docker-devel-consensus-1  | 2023-12-13 22:35:38.062 INFO  - Loading deposit tree snapshot from https://holesky.beaconstate.inf/eth/v1/beacon/deposit_snapshot
eth-docker-devel-consensus-1  | 2023-12-13 22:35:38.071 INFO  - Syncing     *** Target slot: 550378, Head slot: 550371, Waiting for execution layer sync, Connected peers: 0
eth-docker-devel-consensus-1  | 2023-12-13 22:35:38.235 ERROR - Failed to load deposit tree snapshot
eth-docker-devel-consensus-1  | java.io.FileNotFoundException: File 'https://holesky.beaconstate.inf/eth/v1/beacon/deposit_snapshot' not found
eth-docker-devel-consensus-1  | at tech.pegasys.teku.beacon.pow.DepositSnapshotFileLoader.lambda$loadFromUrl$1(DepositSnapshotFileLoader.java:90) ~[teku-beacon-pow-23.12.0.jar:23.12.0]
```

### Steps to Reproduce (Bug)

Deliberately configure an unreachable checkpoint sync url, start a Teku that was synced previously

**Expected behavior:** [What you expect to happen]

If Teku can't reach the checkpoint sync provider for deposit snapshots, it starts without and loads snapshots via the engine API/eth1 provider

The current behavior is brittle and makes the checkpoint sync provider into a single point of failure

### Versions (Add all that apply)
* Software version: 23.12.0

**Comment by rolfyone** [comment]
It looks like this is an issue with the url? should it be `https://holesky.beaconstate.info/eth/v1/beacon/deposit_snapshot`?

do you have `checkpoint-sync-url=https://holesky.beaconstate.inf`?

**Comment by rolfyone** [comment]
Sought clarification incorrect route was to replicate the circumstance and show the concern. We'll need to dig into the details on this one and determine severity.

**Comment by lucassaldanha** [comment]
Thanks again for reporting this issue. I'll look into it. 

In the meantime, if you want to eliminate that single point of failure in your setup, you can use the flag `--deposit-snapshot-enabled` to revert back to the previous behaviour (forcing Teku to use the bundled deposit tree snapshot instead of trying to read if from the Beacon API).

**Comment by lucassaldanha** [comment]
We will revert this behaviour in 23.12.1 and reintroduce it in our next version with a proper fallback mechanism implemented (#7715)

**Comment by lucassaldanha** [comment]
@yorickdowne This was fixed on 23.12.1. Please note that we have reverted the behaviour and will revisit its implementation next year.
